qemu/tests/virtio_fs_daemon_readonly.py

```diff
@@ -0,0 +1,156 @@
+        5. Test the mount dir is readonly.
+
+    :param test: QEMU test object.
+    :param params: Dictionary with the test parameters.
+    :param env: Dictionary with test environment.
+    """
+
+    def create_service(session):
+        if os_type == "windows":
+            error_context.context("Create virtiofs service in guest.", test.log.info)
+
+            driver_name = params["driver_name"]
+
+            session = utils_test.qemu.windrv_check_running_verifier(
+                session, vm, test, driver_name
+            )
+            viofs_svc_name = params["viofs_svc_name"]
+            virtio_fs_utils.create_viofs_service(
+                test, params, session, service=viofs_svc_name
+            )
+        return session
+
+    def delete_service():
+        if os_type == "windows":
+            error_context.context("Delete virtiofs service in guest.", test.log.info)
+            session = vm.wait_for_login()
+            virtio_fs_utils.delete_viofs_serivce(test, params, session)
+            session.close()
+
+    def start_service(session):
+        for fs in params.objects("filesystems"):
+            fs_params = params.object_params(fs)
+
+            fs_target = fs_params["fs_target"]
+            fs_dest = fs_params["fs_dest"]
+
+            if os_type == "linux":
+                utils_misc.make_dirs(fs_dest, session)
+                error_context.context(
+                    "Mount virtiofs target %s to %s inside"
+                    " guest." % (fs_target, fs_dest),
+                    test.log.info,
+                )
+                if not utils_disk.mount(
+                    fs_target, fs_dest, "virtiofs", session=session
+                ):
+                    utils_misc.safe_rmdir(fs_dest, session=session)
+                    test.fail("Failed to mount virtiofs {fs_target}.")
+            else:
+                error_context.context("Start virtiofs service in guest.", test.log.info)
+                debug_log_operation = params.get("debug_log_operation")
+                if debug_log_operation:
+                    session = virtio_fs_utils.operate_debug_log(
+                        test, params, session, vm, debug_log_operation
+                    )
+                virtio_fs_utils.start_viofs_service(test, params, session)
+
+                fs_dest = "%s:" % virtio_fs_utils.get_virtiofs_driver_letter(
+                    test, fs_target, session
+                )
+
+            guest_mnts[fs_target] = fs_dest
+        return session
+
+    def stop_service(session):
+        error_context.context("Stop virtiofs service in guest.", test.log.info)
+
+        if os_type == "linux":
+            for fs_target, fs_dest in guest_mnts.items():
+                utils_disk.umount(fs_target, fs_dest, "virtiofs", session=session)
+                utils_misc.safe_rmdir(fs_dest, session=session)
+        else:
+            if params["viofs_svc_name"] == "WinFSP.Launcher":
```

Comment:
As there is only one fs to share, so we can delete this multi-fs service.

Reply:
Updated.